tornado-drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/OCLKernelScheduler.java

```diff
@@ -67,6 +67,9 @@ private void updateProfiler(long executionPlanId, final int taskEvent, final Tas
             long dispatchValue = meta.getProfiler().getTimer(ProfilerType.TOTAL_DISPATCH_KERNEL_TIME);
             dispatchValue += tornadoKernelEvent.getDriverDispatchTime();
             meta.getProfiler().setTimer(ProfilerType.TOTAL_DISPATCH_KERNEL_TIME, dispatchValue);
+            meta.getProfiler().setTaskPowerUsage(ProfilerType.POWER_USAGE_mW, meta.getId(), deviceContext.getPowerUsage());
+            long powerUsage = meta.getProfiler().getTaskTimer(ProfilerType.POWER_USAGE_mW, meta.getId());
+            meta.getProfiler().setTimer(ProfilerType.POWER_USAGE_mW, powerUsage);
```

Comment:
Shall I add a condition here to assess if the device is NVIDIA GPU for OpenCL?

Otherwise, it will be returning zero as `POWER_USAGE_mW`. I tested that with my macOS. What do you think?

Reply:
Returning `0` is confusing. We can add a condition and add it only for the supported platforms.

Reply:
I added such a condition.